### PR TITLE
Rav 4819 log support for synapse

### PR DESCRIPTION
**Pull request by saranshmanglani**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/sql/dialect/MssqlSqlDialect.java [hunk 1]
```diff
@@ -111,12 +112,34 @@ public MssqlSqlDialect(Context context) {
...
   }
 
+  public void unparseOtherFunction(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
+    switch (call.getOperator().getName()) {
+    case "LN":
+    case "LOG10":
+      final String frameName = call.getOperator().getName()
+                                    .equalsIgnoreCase("LN") ? "LOG" : "LOG10";
+      final SqlWriter.Frame logFrame = writer.startFunCall(frameName);
+      if (call.operand(0).getKind().toString().equalsIgnoreCase("CAST")) {
```
Comment (roy-sounak) [inline core/src/main/java/org/apache/calcite/sql/dialect/MssqlSqlDialect.java:131]: Is it possible to unparse 'CAST' as well ?

File: core/src/main/java/org/apache/calcite/sql/dialect/MssqlSqlDialect.java [hunk 2]
```diff
@@ -111,12 +112,34 @@ public MssqlSqlDialect(Context context) {
...
         super.unparseCall(writer, call, leftPrec, rightPrec);
       }
     }
   }
 
+  public void unparseOtherFunction(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
+    switch (call.getOperator().getName()) {
+    case "LN":
+    case "LOG10":
+      final String frameName = call.getOperator().getName()
```
Comment (roy-sounak) [inline core/src/main/java/org/apache/calcite/sql/dialect/MssqlSqlDialect.java:128]: Change var name frameName to operator

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 3]
```diff
@@ -6587,6 +6587,27 @@ public void testCaseExprForEEE() {
...
+  @Test
+  public void testCaseExprForLog10Synapse() {
+    final String query = "SELECT LOG10(\"product_id\") as dd from \"product\"";
+    final String expectedSynapse = "SELECT LOG10([product_id]) AS [DD]"
+            + "\nFROM [foodmart].[product]";
+    sql(query)
+            .withMssql()
+            .ok(expectedSynapse);
+  }
+
```
Comment (roy-sounak) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:6610]: Add a test case for 'Cast' as well

**Review by roy-sounak — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/apache/calcite/sql/dialect/MssqlSqlDialect.java (hunk 1 above)
Comment (saranshmanglani) [reply core/src/main/java/org/apache/calcite/sql/dialect/MssqlSqlDialect.java]: Done

File: core/src/main/java/org/apache/calcite/sql/dialect/MssqlSqlDialect.java (hunk 2 above)
Comment (saranshmanglani) [reply core/src/main/java/org/apache/calcite/sql/dialect/MssqlSqlDialect.java]: Done

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java (hunk 3 above)
Comment (saranshmanglani) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java]: Not Possible. There is no such table in calcite which has varchar type column with numeric values.

**Review by roy-sounak — APPROVED** [review]
LGTM!

**Comment by roy-sounak** [comment]
Approved. Requesting For L2

**Review by soma-mondal — APPROVED** [review]

**Review by deepakDatametica — APPROVED** [review]

**Pull request merged**